src/WaterPokemon.java

```diff
@@ -0,0 +1,88 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class WaterPokemon extends Pokemon{
+    private final List<String> attacks;
+
+    public WaterPokemon(String name, int level, int healthPoints, String food, String speech) {
+        super(name, level, healthPoints,"water",speech, food);
+        this.attacks = setAttacks();
+    }
+
+    private List<String> setAttacks() {
+        List<String> attacks = new ArrayList<>();
+        attacks.add("surf");
+        attacks.add("hydropump");
+        attacks.add("hydrocanon");
+        return attacks;
+    }
+    public List<String> getAttacks() {
+        return attacks;
+    }
+
+    private int attackMultiplier(int attack, Pokemon defender) {
+        attack = switch (defender.getType()) {
+            case "fire" -> attack * 2;
+            case "electric" -> (int) (attack * 1.5);
+            case "grass" -> (int) (attack * 1.1);
+            case "water" -> attack / 2;
+            default -> 15;
+        };
+
+        return attack;
+    }
+
+    private void attackDisplay(Pokemon attacker, Pokemon defender, int damageCount, String attackName) {
```

Comment:
En ook deze helper methode is een goed idee. Maar bij deze methode is de uitwerking exact hetzelfde voor elke subklasse. Daarom had deze methode beter in de Pokemon (super)klasse gekunt.